Tests/tests/test_MotionDetection.cpp

```diff
@@ -0,0 +1,202 @@
+        EXPECT_EQ(response,  string("{\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, armErrmotiondetected)
+{
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("arm"), _T("{\"index\":\"FP_MD\",\"mode\":\"10\" }"), response));
+        EXPECT_EQ(response,  string("{\"success\":false}"));
+}
+
+
+TEST_F(MotionDetectionEventTest, disarm)
+{
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_MD\"}"), response));
+        EXPECT_EQ(response,  string("{\"success\":true}"));
+}
+TEST_F(MotionDetectionEventTest, disarm)
+{
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_MD\"}"), response));
+        EXPECT_EQ(response,  string("{\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, disarmError)
+{
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_XX\"}"), response));
+        EXPECT_EQ(response,  string("{\"success\":false}"));
+}
+
+
+TEST_F(MotionDetectionEventTest, isarmed)
+{
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("isarmed"), _T("{\"index\"=\"FP_MD\"}"), response));
+        EXPECT_EQ(response,  string("{\"state\":false,\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, setNoMotionPeriod)
+{
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setNoMotionPeriod"), _T("{\"index\":\"FP_MD\",\"period\":\"10\"}"), response));
+        EXPECT_EQ(response,  string("{\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, getNoMotionPeriod)
+{
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getNoMotionPeriod"), _T("{\"index\"=\"FP_MD\"}"), response));
+        EXPECT_EQ(response,  string("{\"period\":\"0\",\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, setSensitivity)
+{
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setSensitivity"), _T("{\"index\":\"FP_MD\",\"name\":\"high\",\"value\":\"2\"}"), response));
+        EXPECT_EQ(response,  string("{\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, getSensitivity)
+{
+          ON_CALL(motionDetectionImplMock, MOTION_DETECTION_GetSensitivity(::testing::_,::testing::_,::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index, char** sensitivity, int* currentMode) {
+                *currentMode = 1;
+                *sensitivity = (char *)malloc(sizeof(STR_SENSITIVITY_LOW));
+                memset(*sensitivity, 0, sizeof(STR_SENSITIVITY_LOW));
+                strcpy(*sensitivity, STR_SENSITIVITY_LOW);
+                return MOTION_DETECTION_RESULT_SUCCESS;
+            }));
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getSensitivity"), _T("{\"index\":\"FP_MD\"}"), response));
+        EXPECT_EQ(response,  string("{\"value\":\"high\",\"success\":true}"));
+}
```

Comment:
I don't understand the idea of this test
you invoke "getSensitivity" to get some value
in the mock you set value to LOW
but in the response it is "high"
does it make any sense?